lib/reverse_sentence.rb

```diff
@@ -1,6 +1,36 @@
+class String
+  def peek
+    if self[length.pred] == ' '
+      match(/\s+$/)[0].delete_suffix ' '
+    else
+      match(/[a-zA-Z\.,'\?!]+$/)[0]
+    end
+  end
+
+  def rotate! i
+    suffix = peek
+    
+    insert i, (suffix + ' ')
```

Comment:
It's neat that you're inserting these methods into the String class  However you should be aware of the overhead these methods inflict.  I'm pretty sure that insert, and delete_suffix are going to be O(n) methods.

Since you're doing this in a loop...